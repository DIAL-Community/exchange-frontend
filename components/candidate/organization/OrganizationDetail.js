import { useEffect, useRef, useState } from 'react'
import { useApolloClient, useQuery } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../../lib/apolloClient'
import Breadcrumb from '../../shared/Breadcrumb'
import { handleLoadingQuery, handleMissingData, handleQueryError } from '../../shared/GraphQueryHandler'
import {
  CANDIDATE_ORGANIZATION_DETAIL_QUERY, CANDIDATE_ORGANIZATION_POLICY_QUERY
} from '../../shared/query/candidateOrganization'
import { fetchOperationPolicies } from '../../utils/policy'
import OrganizationDetailLeft from './OrganizationDetailLeft'
import OrganizationDetailRight from './OrganizationDetailRight'

const OrganizationDetail = ({ slug }) => {
  const scrollRef = useRef(null)
  const client = useApolloClient()

  const [editingAllowed, setEditingAllowed] = useState(false)

  const { loading, error, data, refetch } = useQuery(CANDIDATE_ORGANIZATION_DETAIL_QUERY, {
    variables: { slug },
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.VIEWING
      }
    }
  })

  useEffect(() => {
    fetchOperationPolicies(
      client,
      CANDIDATE_ORGANIZATION_POLICY_QUERY,
      ['editing']
    ).then(policies => {
      setEditingAllowed(policies['editing'])
    })
  }, [client])

  if (loading) {
    return handleLoadingQuery()
  } else if (error) {
    return handleQueryError(error)
  } else if (!data?.candidateOrganization) {
    return handleMissingData()
  }

  const { candidateOrganization: organization } = data

  const slugNameMapping = () => {
    const map = {}
    map[organization.slug] = organization.name

    return map
  }

  return (
    <div className='lg:px-8 xl:px-56 flex flex-col text-dial-stratos '>
      <div className='px-4 lg:px-6 py-4 bg-dial-violet ribbon-detail z-40'>
        <Breadcrumb slugNameMapping={slugNameMapping()}/>
      </div>
      <div className='flex flex-col lg:flex-row gap-x-8'>
        <div className='lg:basis-1/3 shrink-0'>
          <OrganizationDetailLeft
            organization={organization}
            scrollRef={scrollRef}
          />
        </div>
        <div className='lg:basis-2/3'>
          <OrganizationDetailRight
            ref={scrollRef}
            refetch={refetch}
            organization={organization}
            editingAllowed={editingAllowed}
          />
        </div>
      </div>
    </div>
  )
}

export default OrganizationDetail
