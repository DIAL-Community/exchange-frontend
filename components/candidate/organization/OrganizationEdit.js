import { useCallback } from 'react'
import { useIntl } from 'react-intl'
import { useQuery } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../../lib/apolloClient'
import Breadcrumb from '../../shared/Breadcrumb'
import { handleLoadingQuery, handleMissingData, handleQueryError } from '../../shared/GraphQueryHandler'
import { CANDIDATE_ORGANIZATION_DETAIL_QUERY } from '../../shared/query/candidateOrganization'
import OrganizationForm from './fragments/OrganizationForm'
import OrganizationEditLeft from './OrganizationEditLeft'

const OrganizationEdit = ({ slug }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { loading, error, data } = useQuery(CANDIDATE_ORGANIZATION_DETAIL_QUERY, {
    variables: { slug },
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.EDITING
      }
    }
  })

  if (loading) {
    return handleLoadingQuery()
  } else if (error) {
    return handleQueryError(error)
  } else if (!data?.candidateOrganization) {
    return handleMissingData()
  }

  const { candidateOrganization: organization } = data

  const slugNameMapping = () => {
    const map = {
      edit: format('app.edit')
    }
    map[organization.slug] = organization.name

    return map
  }

  return (
    <div className='lg:px-8 xl:px-56 flex flex-col'>
      <div className='px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40'>
        <Breadcrumb slugNameMapping={slugNameMapping()}/>
      </div>
      <div className='flex flex-col lg:flex-row gap-x-8'>
        <div className='lg:basis-1/3 shrink-0'>
          <OrganizationEditLeft organization={organization} />
        </div>
        <div className='lg:basis-2/3'>
          <OrganizationForm organization={organization} />
        </div>
      </div>
    </div>
  )
}

export default OrganizationEdit
