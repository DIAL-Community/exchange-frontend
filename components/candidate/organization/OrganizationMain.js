import OrganizationMainLeft from './OrganizationMainLeft'
import OrganizationMainRight from './OrganizationMainRight'

const OrganizationMain = ({ activeTab }) => {
  return (
    <div className='px-4 lg:px-8 xl:px-24 3xl:px-56'>
      <div className='grid grid-cols-3 gap-x-8'>
        <div className='hidden md:block col-span-1'>
          <OrganizationMainLeft activeTab={activeTab} />
        </div>
        <div className='col-span-3 md:col-span-2'>
          <OrganizationMainRight activeTab={activeTab} />
        </div>
      </div>
    </div>
  )
}

export default OrganizationMain
