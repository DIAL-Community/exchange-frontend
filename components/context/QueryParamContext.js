import { createContext, useState } from 'react'

const QueryParamContext = createContext()
const QueryParamContextProvider = ({ children }) => {
  const [interactionDetected, setInteractionDetected] = useState(false)
  return (
    <QueryParamContext.Provider value={{ ...{ interactionDetected, setInteractionDetected } }}>
      {children}
    </QueryParamContext.Provider>
  )
}

export { QueryParamContextProvider, QueryParamContext }
