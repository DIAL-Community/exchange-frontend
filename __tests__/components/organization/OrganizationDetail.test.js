import { screen } from '@testing-library/dom'
import userEvent from '@testing-library/user-event'
import { QueryParamContextProvider } from '../../../components/context/QueryParamContext'
import OrganizationEdit from '../../../components/organization/OrganizationEdit'
import { QueryErrorCode } from '../../../components/shared/GraphQueryHandler'
import { CREATE_ORGANIZATION } from '../../../components/shared/mutation/organization'
import { COMMENTS_QUERY } from '../../../components/shared/query/comment'
import {
  ORGANIZATION_DETAIL_QUERY, ORGANIZATION_PAGINATION_ATTRIBUTES_QUERY, PAGINATED_ORGANIZATIONS_QUERY
} from '../../../components/shared/query/organization'
import { render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { mockNextAuthUseSession, mockNextUseRouter, mockTenantApi } from '../../utils/nextMockImplementation'
import { commentsQuery, createOrganization, organizationDetail } from './data/OrganizationDetail.data'
import { organizationPaginationAttribute, paginatedOrganizations } from './data/OrganizationMain.data'

mockTenantApi()
mockNextUseRouter()
describe('Unit tests for the organization detail page.', () => {
  const mockOrganization = generateMockApolloData(
    ORGANIZATION_DETAIL_QUERY,
    {
      'slug': 'ai4gov'
    },
    null,
    organizationDetail
  )

  const mockOrganizationComments = generateMockApolloData(
    COMMENTS_QUERY,
    {
      'commentObjectId': 190,
      'commentObjectType': 'ORGANIZATION'
    },
    null,
    commentsQuery
  )

  test('Should render detail of a organization.', async () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockOrganization, mockOrganizationComments]}>
        <QueryParamContextProvider>
          <OrganizationEdit slug='ai4gov' />
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('AI4GOV')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('Should render unauthorized for non logged in user.', async () => {
    const graphQueryErrors = {
      graphQueryErrors: [{
        'message': 'Viewing is not allowed.',
        'locations': [
          {
            'line': 2,
            'column': 3
          }
        ],
        'path': [
          'buildingBlock'
        ],
        'extensions': {
          'code': QueryErrorCode.UNAUTHORIZED
        }
      }]
    }
    const mockOrganizationPolicyQueryError = generateMockApolloData(
      ORGANIZATION_DETAIL_QUERY,
      {
        'slug': 'ai4gov'
      },
      graphQueryErrors,
      null
    )
    const { container } = render(
      <CustomMockedProvider mocks={[mockOrganizationPolicyQueryError, mockOrganizationComments]}>
        <QueryParamContextProvider>
          <OrganizationEdit slug='ai4gov' />
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should render edit page for logged in user.', async () => {
    mockNextAuthUseSession()

    const mockCreateBuildingBlock = generateMockApolloData(
      CREATE_ORGANIZATION,
      {
        'name': 'AI4GOV - Edited',
        'slug': 'ai4gov',
        'aliases': [
          ''
        ],
        'website': 'www.ai4gov.net',
        'isEndorser': true,
        'whenEndorsed': null,
        'endorserLevel': 'none',
        'isMni': false,
        'description': 'Description for the organization.',
        'hasStorefront': false
      },
      null,
      createOrganization
    )

    const mockOrganizationPaginationAttribute = generateMockApolloData(
      ORGANIZATION_PAGINATION_ATTRIBUTES_QUERY,
      { search: '' },
      null,
      organizationPaginationAttribute
    )

    const mockPaginatedOrganizations = generateMockApolloData(
      PAGINATED_ORGANIZATIONS_QUERY,
      { search: '', limit: 8, offset: 0 },
      null,
      paginatedOrganizations
    )

    const { container } = render(
      <CustomMockedProvider
        mocks={[
          mockOrganization,
          mockOrganizationComments,
          mockCreateBuildingBlock,
          mockOrganizationPaginationAttribute,
          mockPaginatedOrganizations,
          mockOrganization
        ]}
      >
        <QueryParamContextProvider>
          <OrganizationEdit slug='ai4gov' />
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('AI4GOV')).toBeInTheDocument()

    const repositoryNameInput = screen.getByDisplayValue('AI4GOV')
    expect(repositoryNameInput.value).toBe('AI4GOV')

    const user = userEvent.setup()
    await user.type(repositoryNameInput, ' - Edited')
    expect(repositoryNameInput.value).toBe('AI4GOV - Edited')

    const repositorySubmitButton = screen.getByText('Submit Organization')
    await user.click(repositorySubmitButton)

    expect(container).toMatchSnapshot()
  })
})
