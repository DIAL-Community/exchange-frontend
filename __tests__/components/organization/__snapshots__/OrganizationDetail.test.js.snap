// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the organization detail page. Should render detail of a organization. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 lg:h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-plum font-semibold"
              >
                AI4GOV
              </div>
              <div
                class="flex justify-center items-center py-8 px-4 bg-white rounded border"
              >
                <div
                  class="inline"
                >
                  <img
                    alt="Logo of Organization."
                    class="object-contain"
                    src="http://localhost:3000//assets/organizations/ai4gov.png"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              >
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="font-semibold text-dial-sapphire"
                  >
                    Website
                  </div>
                  <div
                    class="flex text-dial-stratos"
                  >
                    <a
                      class="flex border-b border-dial-iris-blue "
                      href="//www.ai4gov.net"
                      rel="noreferrer"
                      target="_blank"
                    >
                      <div
                        class="line-clamp-1 break-all"
                      >
                        www.ai4gov.net
                      </div>
                    </a>
                     ⧉
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="flex flex-row gap-3"
                  >
                    <div
                      class="font-semibold text-dial-sapphire"
                    >
                      Sectors
                    </div>
                    <div
                      class="flex gap-3 ml-auto"
                    />
                  </div>
                  <div
                    class="flex flex-col gap-y-2"
                  >
                    <div>
                      <div>
                        Digital, Data, and Technology
                      </div>
                    </div>
                    <div>
                      <div>
                        Education and Social Development
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <div
          class="h-[60vh]"
        >
          <div
            class="flex bg-dial-alice-blue h-full"
          >
            <div
              class="text-dial-stratos text-lg w-full my-auto flex flex-col gap-4"
            >
              <svg
                class="mx-auto"
                fill="currentColor"
                height="3em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="3em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm0-384c13.3 0 24 10.7 24 24V264c0 13.3-10.7 24-24 24s-24-10.7-24-24V152c0-13.3 10.7-24 24-24zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
                />
              </svg>
              <div
                class="text-center mt-5"
              >
                You are not authorized to view this page
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the organization detail page. Should render edit page for logged in user. 1`] = `
<div>
  <div
    class="Toastify"
  >
    <div
      class="Toastify__toast-container Toastify__toast-container--top-center"
      style="width: auto;"
    >
      <div
        class="Toastify__toast Toastify__toast-theme--light Toastify__toast--success Toastify--animate Toastify__slide-enter--top-center"
        data-in="true"
        id="toastId"
      >
        <div
          class="Toastify__toast-body"
          role="alert"
        >
          <div
            class="Toastify__toast-icon Toastify--animate-icon Toastify__zoom-enter"
          >
            <svg
              fill="var(--toastify-icon-color-success)"
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
            >
              <path
                d="M12 0a12 12 0 1012 12A12.014 12.014 0 0012 0zm6.927 8.2l-6.845 9.289a1.011 1.011 0 01-1.43.188l-4.888-3.908a1 1 0 111.25-1.562l4.076 3.261 6.227-8.451a1 1 0 111.61 1.183z"
              />
            </svg>
          </div>
          <div>
            Organization submitted successfully.
          </div>
        </div>
        <button
          aria-label="close"
          class="Toastify__close-button Toastify__close-button--light"
          type="button"
        >
          <svg
            aria-hidden="true"
            viewBox="0 0 14 16"
          >
            <path
              d="M7.71 8.23l3.75 3.75-1.48 1.48-3.75-3.75-3.75 3.75L1 11.98l3.75-3.75L1 4.48 2.48 3l3.75 3.75L9.98 3l1.48 1.48-3.75 3.75z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <div
          class="Toastify__progress-bar--wrp"
          data-hidden="false"
        >
          <div
            class="Toastify__progress-bar--bg Toastify__progress-bar-theme--light Toastify__progress-bar--success"
          />
          <div
            aria-hidden="false"
            aria-label="notification timer"
            class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar-theme--light Toastify__progress-bar--success"
            role="progressbar"
            style="animation-duration: 1000ms; animation-play-state: paused;"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 lg:h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-plum font-semibold"
              >
                AI4GOV -- Edited
              </div>
              <div
                class="flex justify-center items-center py-8 px-4 bg-white rounded border"
              >
                <div
                  class="inline"
                >
                  <img
                    alt="Logo of Organization."
                    class="object-contain"
                    src="http://localhost:3000//assets/organizations/ai4gov.png"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              >
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="font-semibold text-dial-sapphire"
                  >
                    Website
                  </div>
                  <div
                    class="flex text-dial-stratos"
                  >
                    <a
                      class="flex border-b border-dial-iris-blue "
                      href="//www.ai4gov.net"
                      rel="noreferrer"
                      target="_blank"
                    >
                      <div
                        class="line-clamp-1 break-all"
                      >
                        www.ai4gov.net
                      </div>
                    </a>
                     ⧉
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="flex flex-row gap-3"
                  >
                    <div
                      class="font-semibold text-dial-sapphire"
                    >
                      Sectors
                    </div>
                    <div
                      class="flex gap-3 ml-auto"
                    >
                      <button
                        class="cursor-pointer bg-dial-iris-blue px-2 py-1 rounded text-white"
                        type="button"
                      >
                        <svg
                          class="inline pb-0.5"
                          fill="none"
                          height="1em"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M12 20h9"
                          />
                          <path
                            d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
                          />
                        </svg>
                        <span
                          class="text-sm px-1"
                        >
                          Edit
                        </span>
                      </button>
                    </div>
                  </div>
                  <div
                    class="flex flex-col gap-y-2"
                  >
                    <div>
                      <div>
                        Digital, Data, and Technology
                      </div>
                    </div>
                    <div>
                      <div>
                        Education and Social Development
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <form>
          <div
            class="px-4 lg:px-0 py-4 lg:py-6 text-dial-plum"
          >
            <div
              class="flex flex-col gap-y-6 text-sm"
            >
              <div
                class="text-xl font-semibold"
              >
                Edit AI4GOV -- Edited
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire required-field"
                  for="name"
                >
                  Name
                </label>
                <input
                  class="w-full text-sm"
                  id="name"
                  name="name"
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire"
                >
                  Aliases
                </label>
                <div
                  class="flex gap-x-2"
                >
                  <input
                    class="w-full text-sm"
                    name="aliases.0.value"
                    placeholder="Alias"
                    type="text"
                    value=""
                  />
                  <span>
                    <button
                      class="bg-dial-sapphire shadow-md px-3 py-2.5 hover:bg-dial-angel"
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                        />
                      </svg>
                    </button>
                  </span>
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire required-field"
                  for="website"
                >
                  Website
                </label>
                <div
                  class="border border-transparent url-input bg-white flex items-center"
                >
                  <span
                    class="select-none text-dial-gray"
                  >
                    https://
                  </span>
                  <input
                    class="url w-full text-sm placeholder-dial-gray border-0 py-0 px-0 focus:outline-0"
                    id="website"
                    placeholder="Website"
                    value="www.ai4gov.net"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire"
                >
                  Logo
                </label>
                <label
                  class="flex items-center"
                >
                  <input
                    accept=".jpg,.jpeg,.png,.bmp"
                    class="bg-white file:hidden w-full text-sm"
                    name="imageFile"
                    type="file"
                  />
                  <div
                    class="-ml-11 flex p-1"
                  >
                    <span
                      class="h-full pl-1 border-l text-dial-gray-dark border-dial-gray-dark"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        version="1.1"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13 14h-10v-2l3-5 4.109 5 2.891-2v4z"
                        />
                        <path
                          d="M13 7.5c0 0.828-0.672 1.5-1.5 1.5s-1.5-0.672-1.5-1.5 0.672-1.5 1.5-1.5c0.828 0 1.5 0.672 1.5 1.5z"
                        />
                        <path
                          d="M14.341 3.579c-0.347-0.473-0.831-1.027-1.362-1.558s-1.085-1.015-1.558-1.362c-0.806-0.591-1.197-0.659-1.421-0.659h-7.75c-0.689 0-1.25 0.561-1.25 1.25v13.5c0 0.689 0.561 1.25 1.25 1.25h11.5c0.689 0 1.25-0.561 1.25-1.25v-9.75c0-0.224-0.068-0.615-0.659-1.421zM12.271 2.729c0.48 0.48 0.856 0.912 1.134 1.271h-2.406v-2.405c0.359 0.278 0.792 0.654 1.271 1.134zM14 14.75c0 0.136-0.114 0.25-0.25 0.25h-11.5c-0.135 0-0.25-0.114-0.25-0.25v-13.5c0-0.135 0.115-0.25 0.25-0.25 0 0 7.749-0 7.75 0v3.5c0 0.276 0.224 0.5 0.5 0.5h3.5v9.75z"
                        />
                      </svg>
                    </span>
                  </div>
                </label>
                <div
                  class="text-xs italic text-dial-stratos"
                >
                  Note: Supported file types are jpg, png, and bmp
                </div>
              </div>
              <label
                class="flex gap-x-2 items-center self-start text-dial-sapphire"
              >
                <input
                  class="h-5 w-5"
                  name="isEndorser"
                  type="checkbox"
                />
                Organization is an endorser of the Principle for Digital Development.
              </label>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire"
                >
                  Endorsed Date
                </label>
                <input
                  class="w-full text-sm"
                  name="whenEndorsed"
                  placeholder="Endorsed Date"
                  type="date"
                  value=""
                />
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire"
                >
                  Endorser Level
                </label>
                <div
                  class=" css-sqp82o-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-:r0:-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class="react-select__control css-afrqkl-control"
                  >
                    <div
                      class="react-select__value-container react-select__value-container--has-value css-17t7gkf-ValueContainer"
                    >
                      <div
                        class="react-select__single-value css-1dimb5e-singleValue"
                      >
                        None
                      </div>
                      <div
                        class="react-select__input-container css-qbdosj-Input"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class="react-select__input"
                          id="react-select-id-endorserLevel"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <label
                class="flex gap-x-2 items-center self-start text-dial-sapphire"
              >
                <input
                  class="h-5 w-5"
                  name="isMni"
                  type="checkbox"
                />
                Organization is a mobile network integrator organization.
              </label>
              <label
                class="flex gap-x-2 items-center self-start text-dial-sapphire"
              >
                <input
                  class="h-5 w-5"
                  name="hasStorefront"
                  type="checkbox"
                />
                Organization has a storefront.
              </label>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire required-field"
                >
                  Description
                </label>
                <div
                  class="relative"
                >
                  <svg
                    class="absolute text-lg inset-x-1/2 top-10 spinner"
                    fill="currentColor"
                    height="2em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="2em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
                    />
                  </svg>
                  <div
                    class="html-editor"
                  >
                    <textarea
                      id="description-editor"
                      style="visibility: hidden;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="flex flex-wrap text-base mt-6 gap-3"
              >
                <button
                  class="submit-button"
                  disabled=""
                  type="submit"
                >
                  Submit Organization
                  <svg
                    class="spinner ml-3"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
                    />
                  </svg>
                </button>
                <button
                  class="cancel-button"
                  disabled=""
                  type="button"
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the organization detail page. Should render unauthorized for non logged in user. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 lg:h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-plum font-semibold"
              >
                AI4GOV
              </div>
              <div
                class="flex justify-center items-center py-8 px-4 bg-white rounded border"
              >
                <div
                  class="inline"
                >
                  <img
                    alt="Logo of Organization."
                    class="object-contain"
                    src="http://localhost:3000//assets/organizations/ai4gov.png"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              >
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="font-semibold text-dial-sapphire"
                  >
                    Website
                  </div>
                  <div
                    class="flex text-dial-stratos"
                  >
                    <a
                      class="flex border-b border-dial-iris-blue "
                      href="//www.ai4gov.net"
                      rel="noreferrer"
                      target="_blank"
                    >
                      <div
                        class="line-clamp-1 break-all"
                      >
                        www.ai4gov.net
                      </div>
                    </a>
                     ⧉
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="flex flex-row gap-3"
                  >
                    <div
                      class="font-semibold text-dial-sapphire"
                    >
                      Sectors
                    </div>
                    <div
                      class="flex gap-3 ml-auto"
                    />
                  </div>
                  <div
                    class="flex flex-col gap-y-2"
                  >
                    <div>
                      <div>
                        Digital, Data, and Technology
                      </div>
                    </div>
                    <div>
                      <div>
                        Education and Social Development
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <div
          class="h-[60vh]"
        >
          <div
            class="flex bg-dial-alice-blue h-full"
          >
            <div
              class="text-dial-stratos text-lg w-full my-auto flex flex-col gap-4"
            >
              <svg
                class="mx-auto"
                fill="currentColor"
                height="3em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="3em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm0-384c13.3 0 24 10.7 24 24V264c0 13.3-10.7 24-24 24s-24-10.7-24-24V152c0-13.3 10.7-24 24-24zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
                />
              </svg>
              <div
                class="text-center mt-5"
              >
                You are not authorized to view this page
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
